py/scripts/generate_imported_types.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import re
+import subprocess
+import sys
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+OPENAPI_SPEC_PATH = os.path.join(SCRIPT_DIR, "../../imported_types.json")
+INTERNAL_TYPES_OUTPUT_PATH = os.path.join(SCRIPT_DIR, "../src/braintrust/_imported_types.py")
+
+
+def generate_internal_types():
+    subprocess.run(
+        [
+            "datamodel-codegen",
+            "--input",
+            OPENAPI_SPEC_PATH,
+            "--input-file-type",
+            "openapi",
+            "--output",
+            INTERNAL_TYPES_OUTPUT_PATH,
+            "--output-model-type",
+            "typing.TypedDict",
+            "--target-python-version",
+            "3.9",
+            "--custom-file-header",
+            '''"""
+Do not import this file directly. See `imported_types.py` for the classes that have a stable API.
+
+Auto-generated file -- do not modify.
+"""''',
+            "--special-field-name-prefix",
+            "",
+            "--enum-field-as-literal",
+            "all",
+            "--capitalize-enum-members",
+            "--use-generic-container-types",
+            "--use-field-description",
+            "--strict-nullable",
+            "--parent-scoped-naming",
+        ],
+        stdout=sys.stderr,
+        check=True,
+    )
+    cleanup_internal_types()
+
+
+def cleanup_internal_types():
+    with open(INTERNAL_TYPES_OUTPUT_PATH, "r") as f:
+        contents = f.read()
+
+    # Replace `NotRequired[...]` with `NotRequired[Optional[...]]` for Python
```

Comment:
These cleanups were copied over from our internal script to auto-generate python typeddicts from an openapi spec.